test/test_utils/test_coalescer.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+
+import pytest
+
+import scipy.sparse
+
+from autoPyTorch.utils.implementations import MinorityCoalescing
+
+
+@pytest.fixture
+def X1():
+    # Generates an array with categories 3, 4, 5, 6, 7 and occurences of 30%,
+    # 30%, 30%, 5% and 5% respectively
+    X = np.vstack((
+        np.ones((30, 10)) * 3,
+        np.ones((30, 10)) * 4,
+        np.ones((30, 10)) * 5,
+        np.ones((5, 10)) * 6,
+        np.ones((5, 10)) * 7,
+    ))
+    for col in range(X.shape[1]):
+        np.random.shuffle(X[:, col])
+    return X
+
+
+@pytest.fixture
+def X2():
+    # Generates an array with categories 3, 4, 5, 6, 7 and occurences of 5%,
+    # 5%, 5%, 35% and 50% respectively
+    X = np.vstack((
+        np.ones((5, 10)) * 3,
+        np.ones((5, 10)) * 4,
+        np.ones((5, 10)) * 5,
+        np.ones((35, 10)) * 6,
+        np.ones((50, 10)) * 7,
+    ))
+    for col in range(X.shape[1]):
+        np.random.shuffle(X[:, col])
+    return X
+
+
+def test_default(X1):
+    X = X1
+    X_copy = np.copy(X)
+    Y = MinorityCoalescing().fit_transform(X)
+    np.testing.assert_array_almost_equal(Y, X_copy)
+    # Assert no copies were made
+    assert id(X) == id(Y)
+
+
+def test_coalesce_10_percent(X1):
+    X = X1
+    Y = MinorityCoalescing(minimum_fraction=.1).fit_transform(X)
+    for col in range(Y.shape[1]):
+        hist = np.histogram(Y[:, col], bins=np.arange(-2, 7))
+        np.testing.assert_array_almost_equal(hist[0], [10, 0, 0, 0, 0, 30, 30, 30])
+    # Assert no copies were made
+    assert id(X) == id(Y)
```

Comment:
If you would like this method to share the address before and after the transformation, we should not return the array as a result, because it just means if we make any changes to the `X`, it means `Y` will also change unexpectedly.
It is not safe.

Reply:
As the name of the method (transform) indicates, it is within the expectation that the input argument will be transformed.

Normally the input can be an array of thousands of elements. Creating a copy every time we transform is not feasible, so we do expect to transform it.

Reply:
Yes, the copy take memory, so I would like it to be a destructive method.
Because we potentially change the original (the one before transformed) and those changes will be reflected to the transformed version as well.
In other words, we do not have to return the transformed version.